gobblin-utility/src/main/java/gobblin/util/JobLauncherUtils.java

```diff
@@ -212,10 +212,10 @@ private static FileSystem getFsWithProxy(State state, String writerFsUri)
 
   private static ParallelRunner getParallelRunner(FileSystem fs, Closer closer, int parallelRunnerThreads,
       Map<String, ParallelRunner> parallelRunners) {
-    String uri = fs.getUri().toString();
-    if (!parallelRunners.containsKey(uri)) {
-      parallelRunners.put(uri, closer.register(new ParallelRunner(parallelRunnerThreads, fs)));
+    String uriAndHomeDir = fs.getUri().toString() + fs.getHomeDirectory();
```

Comment:
We could probably changed this to `new Path(new Path(fs.getUri), fs.getHomeDirectory())`, and make the key for the `Map` a `Path` instead of a `String`.